python/pyspark/sql/session.py

```diff
@@ -221,6 +221,9 @@ def __init__(self, sparkContext, jsparkSession=None):
                 or SparkSession._instantiatedSession._sc._jsc is None:
             SparkSession._instantiatedSession = self
 
+    def _repr_html_(self):
+        return self.sparkContext._repr_html_()
```

Comment:
As @felixcheung suggested I think it might make sense to include some extra Spark SQL specific things.

I think the catalog implementation type here (can be very useful for understanding why hive UDFs are not working) (which you can get from the session config with the `spark.sql.catalogImplementation` key) and the current database (which you can get from the catalog object from the session). The catalog implementation is especially useful since we currently do a "fallback" from hive supported to non-hive supported and the user might not have noticed if they launched in Jupyter where the log messages are a bit more obscure -- something I've been meaning to work on in https://github.com/apache/spark/pull/17298 but I've gotten a bit distracted).

It might also make sense to return a different URL link (e.g. to the SQL page rather than the default page which takes people to the Jobs section) but